url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/inport/StaticImport.java

```diff
@@ -0,0 +1,2125 @@
+			setnames.add(s.getItemset());
+		}
+		for (Helm s : helms) {
+			setnames.add(s.getItemset());
+		}
+		isets = setnames;
+		try {
+			File outd = new File("data" + File.separator);
+			outd.mkdirs();
+			BufferedWriter out = new BufferedWriter(
+					new OutputStreamWriter(new FileOutputStream("data"
+							+ File.separator + "_sets.xml"), "UTF-8"));
+
+			JAXBContext context = JAXBContext.newInstance(ItemSets.class);
+			Marshaller m = context.createMarshaller();
+			m.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+			m.marshal(new ItemSets(setnames), out);
+			out.close();
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	private static void exportPricesCSV() {
+		StringBuilder sb = new StringBuilder();
+
+		sb.append("id;price;Typ;Set;Itemname;Level;Stars;Useful;Armortype;");
+		sb.append("\n");
+
+		for (Shoes s : shoes) {
+			String x = s.getPrice() + "";
+			x = x.replace(".", ",");
+			sb.append(s.getId() + ";" + x + ";Shoes;"
+					+ s.getItemset().getName() + ";" + s.getName() + ";"
+					+ s.getLevel() + ";" + s.getStars() + ";"
+					+ s.getItemset().getUsefulChars() + ";" + s.getTyp());
+			sb.append("\n");
+		}
+		for (Pants s : pants) {
+			String x = s.getPrice() + "";
+			x = x.replace(".", ",");
+			sb.append(s.getId() + ";" + x + ";Pants;"
+					+ s.getItemset().getName() + ";" + s.getName() + ";"
+					+ s.getLevel() + ";" + s.getStars() + ";"
+					+ s.getItemset().getUsefulChars() + ";" + s.getTyp());
+			sb.append("\n");
+		}
+		for (Gloves s : gloves) {
+			String x = s.getPrice() + "";
+			x = x.replace(".", ",");
+			sb.append(s.getId() + ";" + x + ";Gloves;"
+					+ s.getItemset().getName() + ";" + s.getName() + ";"
+					+ s.getLevel() + ";" + s.getStars() + ";"
+					+ s.getItemset().getUsefulChars() + ";" + s.getTyp());
+			sb.append("\n");
+		}
+		for (Armor s : armors) {
+			String x = s.getPrice() + "";
+			x = x.replace(".", ",");
+			sb.append(s.getId() + ";" + x + ";Armor;"
+					+ s.getItemset().getName() + ";" + s.getName() + ";"
+					+ s.getLevel() + ";" + s.getStars() + ";"
+					+ s.getItemset().getUsefulChars() + ";" + s.getTyp());
+			sb.append("\n");
+		}
+		for (Helm s : helms) {
+			String x = s.getPrice() + "";
+			x = x.replace(".", ",");
+			sb.append(s.getId() + ";" + x + ";Helm;" + s.getItemset().getName()
+					+ ";" + s.getName() + ";" + s.getLevel() + ";"
+					+ s.getStars() + ";" + s.getItemset().getUsefulChars()
+					+ ";" + s.getTyp());
+			sb.append("\n");
+		}
+
+		try {
+			File outd = new File("data" + File.separator);
+			outd.mkdirs();
+			BufferedWriter out = new BufferedWriter(
+					new OutputStreamWriter(new FileOutputStream("data"
+							+ File.separator + "prices.csv"), "UTF-8"));
+
+			out.write(sb.toString());
+			out.close();
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	private static void importPrices() {
+		List<String> slist = new ArrayList<>();
+		try {
+			BufferedReader br = new BufferedReader(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *br*. The resource is referenced at lines 2040, 2041, 2044, 2047. The resource is closed at line 2050. There are other execution paths that do not contain closure statements, for example, when BufferedReader.readLine throws an exception. Ensure *br* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).